### PR TITLE
Remove subprocess PID from the pid-list returned by childprocesses.

### PR DESCRIPTION
**Pull request by vinjana**
The usual way to call childProcesses is via $(childProcesses) to get the PIDs into an array. If the subshell pid is really needed, it can also be retrieved from the parent process directly.

### CODE REVIEW

File: resources/analysisTools/roddyTools/wrapInScript.sh [hunk 1]
```diff
@@ -220,10 +220,14 @@ childProcesses() {
...
     declare -a pidList=( $(pstree -a -p $$ | cut -d, -f2 | cut -d" " -f1 | grep -v $$ | sed -r 's/\.*//g') )
+
+    ## To get a clean list of subprocesses we remove the PIDs of the cut, grep, and sed commands and that of the current subshell.
+    local currentSubshell=$(bash -c 'echo $PPID')  ## https://stackoverflow.com/questions/20725925/get-pid-of-current-subshell
     for pid in "${pidList[@]}"; do
-        # Remove the PIDs for (at least) the cat and grep commands.
-        if processesExist "$pid"; then
-            echo "$pid"
+        if [[ "$pid" != "$currentSubshellPid" ]]; then
+            if processesExist "$pid"; then
```
Comment (dankwart-de) [inline resources/analysisTools/roddyTools/wrapInScript.sh]: Please put in [[ ]]

File: resources/analysisTools/roddyTools/wrapInScript.sh [hunk 2]
```diff
@@ -220,10 +220,14 @@ childProcesses() {
...
     # interactive mode. In non-interactive mode, pstree appends a '...' to every non-terminal process ID.
     declare -a pidList=( $(pstree -a -p $$ | cut -d, -f2 | cut -d" " -f1 | grep -v $$ | sed -r 's/\.*//g') )
+
+    ## To get a clean list of subprocesses we remove the PIDs of the cut, grep, and sed commands and that of the current subshell.
+    local currentSubshell=$(bash -c 'echo $PPID')  ## https://stackoverflow.com/questions/20725925/get-pid-of-current-subshell
     for pid in "${pidList[@]}"; do
-        # Remove the PIDs for (at least) the cat and grep commands.
-        if processesExist "$pid"; then
-            echo "$pid"
+        if [[ "$pid" != "$currentSubshellPid" ]]; then
```
Comment (dankwart-de) [inline resources/analysisTools/roddyTools/wrapInScript.sh]: Change to proper name

File: resources/analysisTools/roddyTools/wrapInScript.sh [hunk 3]
```diff
@@ -240,13 +244,13 @@ processesExist() {
 killChildProcesses() {
   local KILLSIG=TERM
   declare -a childProcs=( $(childProcesses) )
-  if [[ ${#childProcs[@]} -gt 0 ]]; then
+  if [[ ${#childProcs[@]} -gt 1 ]]; then  ## This is > 1, because the $() represents another child process, that
```
Comment (dankwart-de) [inline resources/analysisTools/roddyTools/wrapInScript.sh]: Rest of the comment?

**Review by dankwart-de — CHANGES_REQUESTED** [review]

File: resources/analysisTools/roddyTools/wrapInScript.sh (hunk 1 above)
Comment (vinjana) [reply resources/analysisTools/roddyTools/wrapInScript.sh]: Nope, this is neither necessary nor useful here. The processesExist function works like this. The alternative (which would be o.k. for me) would be to return something on STDOUT and test that.

File: resources/analysisTools/roddyTools/wrapInScript.sh (hunk 2 above)
Comment (vinjana) [reply resources/analysisTools/roddyTools/wrapInScript.sh]: oops

File: resources/analysisTools/roddyTools/wrapInScript.sh (hunk 1 above)
Comment (vinjana) [reply resources/analysisTools/roddyTools/wrapInScript.sh]: The main reason I did it like this, is that the actually called program in `processesExist` also just returns != if a process does not exist. Changing this to return a string would not only make code here more complex `[[ $(...) != "yes" ]]` but also there (`if $!; then ... fi`). Just something to consider.

**Review by dankwart-de — APPROVED** [review]

**Pull request merged**